src/wallet/mod.rs

```diff
@@ -115,6 +116,13 @@ pub enum AddressIndex {
     /// caller is untrusted; for example when deriving donation addresses on-demand for a public
     /// web page.
     LastUnused,
+    /// Return the address for the first address in the keychain that has not been used in a received
+    /// transaction. Otherwise return a new address as with [`AddressIndex::New`].
+    ///
+    /// Use with caution, if the wallet has not yet detected an address has been used it could
+    /// return an already used address. This function is primarily meant for making use of addresses earlier
+    /// in the keychain that were infact never used.
+    FirstUnused,
```

Comment:
I feel the caution message is not necessary anymore?? As BDK will not give up used address anymore after this change??

Reply:
I believe this caution is referring to the possibility that of:
1. Get a `LastUnused` address (or FirstUnused)
2. Use this address in some way that BDK doesn't immediately know of. Like in an unbroadcasted txn or an external address you intend to receive money to at some point in the future.
3. Then get an `LastUnused` address again (the same address) and use it for something else.

At later time, the address is reused once the unbroadcasted txn or delayed receive from step-2 occurs.